plugins/src/gcc/imp.rs

```diff
@@ -0,0 +1,1270 @@
+        let res = self.srcpad.push_list(list);
+
+        self.state.lock().unwrap().flow_return = res;
+    }
+
+    fn start_task(&self, bwe: &super::BandwidthEstimator) -> Result<(), glib::BoolError> {
+        let weak_controller = bwe.downgrade();
+        let weak_pad = self.srcpad.downgrade();
+        let clock = gst::SystemClock::obtain();
+
+        bwe.imp().state.lock().unwrap().periodic_push = Some(clock.new_periodic_id(
+            clock.time().unwrap() + dur2ts(*BURST_TIME),
+            dur2ts(*BURST_TIME),
+        ));
+
+        self.srcpad.start_task(move || {
+            let pause = || {
+                if let Some(pad) = weak_pad.upgrade() {
+                    pad.pause_task().unwrap();
+                }
+            };
+            let bwe = weak_controller
+                .upgrade()
+                .expect("bwe destroyed while its srcpad task is still running?");
+
+            let lock_state = || bwe.imp().state.lock().unwrap();
+
+            let periodic_id = match lock_state().periodic_push.take() {
+                Some(id) => id,
+                _ => {
+                    gst::info!(CAT, "Peridic pushes are stop, pausing task");
+                    return pause();
+                }
+            };
+
+            if let (Err(err), _) = periodic_id.wait() {
+                match err {
+                    gst::ClockError::Early => (),
+                    _ => {
+                        gst::error!(CAT, "Got error {err:?} on the clock, pausing task");
+
+                        lock_state().flow_return = Err(gst::FlowError::Flushing);
+
+                        return pause();
+                    }
+                }
+            }
+            let list = {
+                let mut state = lock_state();
+                state.periodic_push = Some(periodic_id);
+                state.create_buffer_list(&bwe)
+            };
+
+            if list.len() > 0 {
+                let _ = bwe.imp().push_list(list);
+            }
+        })?;
+
+        Ok(())
+    }
+
+    fn src_activatemode(
+        &self,
+        _pad: &gst::Pad,
+        bwe: &super::BandwidthEstimator,
+        mode: gst::PadMode,
+        active: bool,
+    ) -> Result<(), gst::LoggableError> {
+        if let gst::PadMode::Push = mode {
+            if active {
+                self.state.lock().unwrap().flow_return = Ok(gst::FlowSuccess::Ok);
+                self.start_task(bwe)?;
+            } else {
+                let mut state = self.state.lock().unwrap();
+                state.flow_return = Err(gst::FlowError::Flushing);
+                self.srcpad.stop_task()?;
+            }
+
+            Ok(())
+        } else {
+            Err(gst::LoggableError::new(
+                *CAT,
+                glib::bool_error!("Unsupported pad mode {mode:?}"),
+            ))
+        }
+    }
+}
+
+#[glib::object_subclass]
+impl ObjectSubclass for BandwidthEstimator {
+    const NAME: &'static str = "RTPGCCBwE";
```

Comment:
```suggestion
    const NAME: &'static str = "GstRTPGCCBwE";
```

Or so maybe so we add some more namespacing